src/NuGet.Services.EndToEnd/Support/PushedPackagesFixture.cs

```diff
@@ -234,40 +282,132 @@ private PackageToPrepare InitializePackage(PackageType packageType)
-                    return new PackageToPrepare(Package.Create(id, SemVer2PrerelVersion));
+                    packageToPrepare = new PackageToPrepare(Package.Create(id, SemVer2PrerelVersion));
+                    break;
 
                 case PackageType.SemVer2PrerelRelisted:
-                    return new PackageToPrepare(Package.Create(id, "1.0.0-alpha.1"),
+                    packageToPrepare = new PackageToPrepare(Package.Create(id, "1.0.0-alpha.1"),
                         unlist: true);
+                    break;
 
                 case PackageType.SemVer1StableUnlisted:
-                    return new PackageToPrepare(
+                    packageToPrepare = new PackageToPrepare(
                         Package.Create(id, "1.0.0"),
                         unlist: true);
+                    break;
 
                 case PackageType.Signed:
-                    return new PackageToPrepare(Package.SignedPackage());
+                    packageToPrepare = new PackageToPrepare(Package.SignedPackage());
+                    break;
+
+                case PackageType.SymbolsPackage:
+                    return await PrepareSymbolsPackageAsync(id, "1.0.0", logger);
 
                 case PackageType.SemVer1Stable:
                 case PackageType.FullValidation:
                 default:
-                    return new PackageToPrepare(Package.Create(id, "1.0.0"));
+                    packageToPrepare = new PackageToPrepare(Package.Create(id, "1.0.0"));
+                    break;
+
             }
+
+            packagesToPrepare.Add(packageToPrepare);
+            return packagesToPrepare;
+        }
+
+        private async Task<List<PackageToPrepare>> PrepareSymbolsPackageAsync(string id, string version, ITestOutputHelper logger)
+        {
+            using (var testDirectory = TestDirectory.Create())
+            {
+                // Copy the symbols project from template, also set the ID, Version appropriately in the properties.
+                var projectPath = CopySymbolsProjectFromTemplate(id, version, testDirectory.FullPath);
+
+                // Build the symbols project with the DotNet
+                var buildCommandResult = await DotNetExeClient.BuildProject(projectPath, logger);
+                if (!string.IsNullOrEmpty(buildCommandResult.Error))
+                {
+                    throw new Exception($"Error building symbols package! Error: {buildCommandResult.Error} Output: {buildCommandResult.Output}");
+                }
+
+                // Build nupkg and snupkg from appropriate files
+                var buildOutput = Path.Combine(testDirectory.FullPath, "bin", "Debug");
+                var dllFiles = Directory.GetFiles(buildOutput, "*.dll");
+                var pdbFiles = Directory.GetFiles(buildOutput, "*.pdb");
+                if (dllFiles.Count() == 0 || pdbFiles.Count() == 0)
+                {
+                    throw new Exception($"Symbols project build failed to create DLL or PDBs");
```

Comment:
Don't throw `Exception`. `InvalidOperationException` is better in this case.

An `Exception` cannot be caught. in a useful way.